content/essentials/event-handling.mdx

```diff
@@ -0,0 +1,76 @@
+# Event handlers
+
+Events handlers play an important role in capturing user interactions and executing the relevant logic based on those interactions.
+
+In Solid, event handlers are prefixed with `on`, to help distinguish them from other properties and attributes in your components.
+They can be written as either `onClick` and `onclick`, catering to different coding styles or project conventions.
+
+```tsx
+<button onClick={handleClick}>Click me</button>
+```
+
+<Aside>
+  **Note**: 
+  Events remain static and are _not_ reactive.
+  This is to avoid the costly process of continuously attaching and detaching listeners.
+  Given that event handlers are invoked like regular functions upon each event trigger, there's no inherent need for them to be reactive.
+  <br />
+  If necessary, you can write your handlers accordingly to call a reactive source:
+
+    ```tsx
+    <div onClick={() => props.handleClick?.()} />
+    ```
+
+</Aside>
+
+## Event delegation
+
+Rather than attaching event listeners to every individual element, Solid uses artificial event delegation. 
+This technique consolidates some event listening at a higher level in the document, allowing events to be captured more effectively on one place instead of many. 
+Supported events such as `click`, `mouseover`, and `keydown` are just a few examples that aare optimized in this way. To view the full list see the [reference section](#delegated-events).
+
+**Note**: If you need to support other casings or want to avoid the event delegation, you can use [on:](/reference/on-and-oncapture) to match event handlers that follows the colon.
+
+## Binding events
+
+Solid binds data to event handlers without incurring the usual overhead of JavaScript's [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) method or additional closures. 
+To achieve this you will have to pass an array to the event handler. Notably, the **second item** of the array will be supplied as the handler's **first argument**.
+
+```jsx
+const handler = (data, event) => {
+   console.log('Data:', data, 'Event:', event);
+}
+
+<button onClick={[handler, 'Hello!']}>Click Me</button>
+```
+
+In this example, the string `'Hello!'` (the second item in the array) is passed as the data parameter (the first argument of the handler function) when the button is clicked.
+
+## Reference
+
+### Delegated Events
+
+You can also view this list in our [source code](https://github.com/ryansolid/dom-expressions/blob/8cdb1f14ec82b944147ebc2a2cb7ce20b6e73fab/packages/dom-expressions/src/constants.js#L97).
```

Comment:
This link will never change. https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/constants.js (see DelegatedEvents) or https://github.com/ryansolid/dom-expressions/blob/main/packages/dom-expressions/src/constants.js#L86 seem more robust to future changes, which is the point of linking I think.